web/client/src/library/components/editor/EditorCode.tsx

```diff
@@ -191,6 +205,8 @@ function CodeEditorDefault({
   }, [dialect])
 
   useEffect(() => {
+    if (isTyping) return
```

Comment:
cursor was jumping to the top of the page when user is typing in the middle of content sync, because content was no longer matching. Instead wait for user to end typing go re-do `update-sync content` flow.